### PR TITLE
Add erc721

### PR DESCRIPTION
**Pull request by 0xAshish**
#25

### CODE REVIEW

File: contracts/proofs/ERC721Validator.sol [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+
+    /*
+      check receipt and data field
+      Receipt -->
+        [0]
+        [1]
+        [2]
+        [3]-> [
+          [child token address, [TRANSFER_EVENT_SIGNATURE, from, to, tokenId], <>],
+          [child token address, [LOG_TRANSFER_EVENT_SIGNATURE,token,from,to], <tokenId,input1,input2,tokenId>]
```
Comment (jdkanani) [inline contracts/proofs/ERC721Validator.sol]: Remove multiple `tokenId` from receipt.

File: contracts/proofs/ERC721Validator.sol [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+
+    return false;
+  }
+
+  function _validateLogTransferEvent(
+    address childToken,
+    address from,
+    address to,
+    uint256 tokenId,
+    RLP.RLPItem[] items // [child token address, [LOG_TRANSFER_EVENT_SIGNATURE,token,from,to], <tokenId,input1,input2,tokenId>]
```
Comment (jdkanani) [inline contracts/proofs/ERC721Validator.sol]: Duplicate `tokenId`

File: contracts/proofs/ERC721Validator.sol (hunk 1 above)
Comment (0xAshish) [reply contracts/proofs/ERC721Validator.sol]: yup, that will be removed once I refactor that withdraw manager code(proofs validation functions).

File: contracts/root/ExitManager.sol [hunk 3]
```diff
@@ -207,13 +204,16 @@ contract ExitManager is RootChainable {
   ) internal {
     // Check that we're exiting a known token.
     require(exitsQueues[_exitObject.token] != address(0));
-
+    bytes32 key;
+    if (depositManager.isERC721(_exitObject.token)) {
+      key = keccak256(_exitObject.token, _exitObject.owner, _exitObject.amountOrTokenId);
```
Comment (jdkanani) [inline contracts/root/ExitManager.sol]: abi.Pack?

File: contracts/root/WithdrawManager.sol [hunk 4]
```diff
@@ -218,7 +216,7 @@ contract WithdrawManager is IManager, ExitManager {
     PlasmaExit memory _exitObject = PlasmaExit({
       owner: msg.sender,
       token: _token,
-      amount: _amount,
+      amountOrTokenId: _amountOrTokenId,
       burnt: false
```
[the remark this answers is not in the document]
Comment (0xAshish) [reply contracts/root/WithdrawManager.sol]: we have fixed this with `(owner => exitId) keccak(token+owner) keccak(token+owner+tokenId)`

**Pull request merged**